modules/axhal/src/lib.rs

```diff
@@ -57,6 +57,9 @@ pub mod irq;
 #[cfg(feature = "paging")]
 pub mod paging;
 
+#[cfg(feature = "alloc")]
+mod dma;
```

Comment:
this is extremely strange, why the dma mod is decorated under `#[cfg(feature = "alloc")]`

Reply:
Dma-api use normal alloc api to alloc Vec.

Reply:
> Dma-api use normal alloc api to alloc Vec.

I just do not get it, generally, the size of dma region is known and fixed, why a `Vec` is needed since it's usually related to dynamically arrays with unknown size.

Reply:
dma use global allocator.

Reply:
maybe we can delete this now?